Library/Homebrew/formula_specialties.rb

```diff
@@ -18,12 +18,20 @@ def self.url(val)
 # This formula serves as the base class for several very similar
 # formulae for Amazon Web Services related tools.
 class AmazonWebServicesFormula < Formula
+
+  depends_on :java
```

Comment:
Dependencies (and most other formula state) aren't inherited by subclasses, so it needs to be declared in each formula.